### PR TITLE
Improve tom bootstrap dialog

### PR DESCRIPTION
**Pull request by nqhhdev**
### Ticket 

Screenshot 2024-05-16 at 13 42 14

### Setup `olm` lib for web: when building localhost

- Push files to folder: `assets/assets/js/package`

#### ⇒ MOBILE 

https://github.com/linagora/twake-on-matrix/assets/99852347/1ec5c1b4-8026-4f9f-a2e0-2e60f355b817

https://github.com/linagora/twake-on-matrix/assets/99852347/9902a003-bf83-447d-aa27-6b0fe77cd163

https://github.com/linagora/twake-on-matrix/assets/99852347/f182ebd0-a366-435e-bb2c-34d934ed6571

#### ⇒ WEB

https://github.com/linagora/twake-on-matrix/assets/99852347/7a6b389e-a573-4b27-811c-8278a443d9b2

please demo the case: user login the first time in web

https://github.com/linagora/twake-on-matrix/assets/99852347/502e7bde-c3dc-4d66-a3e4-993cb9b97575

Dialog overlap when resizing screen

https://github.com/linagora/twake-on-matrix/assets/99852347/f4ff4f34-4d02-4870-8a36-ecde1abc56b1

### CODE REVIEW

File: lib/pages/bootstrap/init_client_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  @override
+  void initState() {
+    _initial();
+    Matrix.of(context).onClientLoginStateChanged.stream.listen(
+          _listenClientLoginStateChanged,
+        );
+    WidgetsBinding.instance.addPostFrameCallback(
+      (_) async {
+        _startLoginSSOProgress();
+        await widget
```
Comment (sherlockvn) [inline lib/pages/bootstrap/init_client_dialog.dart:39]: dispose the streamSubscription plz when not used

**Comment by hoangdat** [comment]
please demo the case: user login the first time in web

**Comment by nqhhdev** [comment]
> * [x]  please demo the case: user login the first time in web

done

**Review by Te-Z — APPROVED** [review]

**Comment by sherlockvn** [comment]
https://github.com/linagora/twake-on-matrix/assets/43041967/ca830903-b181-4808-a85b-caa370722a61

Should block user to back when setting up

**Comment by nqhhdev** [comment]
> Screen.Recording.2024-05-17.at.14.29.15.mov 
> Should block user to back when setting up

Fixed on commit 4c12754a633862588b84fc9a057dbc5ae04f632b

**Comment by hoangdat** [comment]
please priority for this sprint @Te-Z

**Review by sherlockvn — APPROVED** [review]

**Comment by sherlockvn** [comment]
- Worked well in my case

**Review by hoangdat — APPROVED** [review]

**Review by hieutbui — APPROVED** [review]

**Comment by Te-Z** [comment]
I can't build your branch on chrome. Did you change something ?

**Comment by nqhhdev** [comment]
> I can't build your branch on chrome. Did you change something ?

At the time you tested, my branch had not yet been rebase to resolve the issue related to building the web for flutter version 3.22.0. (PR fixed issue: #1840).

Now I have rebased it to the latest main branch, you can test it again to make sure it can be merged

**Comment by Te-Z** [comment]
Hi @nqhhdev I have an infinite loading on Chrome

[Capture vidéo du 04-06-2024 08:59:05.webm](https://github.com/linagora/twake-on-matrix/assets/31937920/40c4806f-1ffe-4303-922e-221c5ca7ab33)

**Comment by Te-Z** [comment]
Works fine on Android but there's en error when I try to log in with SSO. I need to close it, then try again. I don't know if it's related. Do you have an idea @nqhhdev @hoangdat ?

**Comment by Te-Z** [comment]
@sherlockvn can you help me check if it works on iOS please ?

**Comment by nqhhdev** [comment]
> Hi @nqhhdev I have an infinite loading on Chrome
> 
>  Capture.video.du.04-06-2024.08.59.05.webm

Still have and infinite loading?

**Comment by sherlockvn** [comment]
> @sherlockvn can you help me check if it works on iOS please ?

It works in my case, but the linagora homeserver take very long time, not related to the PR i think, because last time i checked its pretty smooth

File: lib/pages/bootstrap/tom_bootstrap_dialog.dart [hunk 2]
```diff
@@ -1,112 +1,184 @@
...
+    await widget.client.accountDataLoading;
+    if (widget.client.userID != null) {
+      await setupAdditionalDioCacheOption(widget.client.userID!);
     }
+    setState(() {
+      _uploadRecoveryKeyState = UploadRecoveryKeyState.checkingRecoveryWork;
+    });
+    await _getRecoveryKeyState();
+    bootstrap =
+        widget.client.encryption!.bootstrap(onUpdate: (_) => setState(() {}));
```
Comment (Te-Z) [inline lib/pages/bootstrap/tom_bootstrap_dialog.dart:94]: Causes null pointer exception on my computer. Please avoid using exclamation mark without checking if the value is null or not.

Capture d’écran du 2024-06-06 10-32-00

Comment (nqhhdev) [reply lib/pages/bootstrap/tom_bootstrap_dialog.dart]: I fixed it on last fixup commit, pls verify it again

**Comment by hieutbui** [comment]
I built and tested. It works well on Windows.

**Review by hieutbui — APPROVED** [review]

**Review by Te-Z — APPROVED** [review]

**Pull request merged**